src/coreclr/src/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -437,6 +454,7 @@ protected ComputedInstanceFieldLayout ComputeAutoFieldLayout(MetadataType type,
             var layoutMetadata = type.GetClassLayout();
 
             int packingSize = ComputePackingSize(type, layoutMetadata);
+            packingSize = Math.Min(type.Context.Target.MaximumAutoLayoutPackingSize, packingSize);
```

Comment:
Nit: could you move `TypeSystemContext context = type.Context;` up and use it here?

Reply:
Done. I also addressed several other locations within the function which had the same problem